File: UseCases/src/test/resources/stub/PropertyProtocolTest.swift
var stubbedReadWrite: String! = ""
var readWrite: String {
set {
}
get {
return stubbedReadWrite
}
}
var stubbedReadOnly: Int! = 0
var readOnly: Int {
return stubbedReadOnly
}
var stubbedOptional: UInt!
var optional: UInt? {
set {
}
get {
return stubbedOptional
}
}
var stubbedUnwrapped: String!
var unwrapped: String! {
set {
}
get {
return stubbedUnwrapped
}
}
var stubbedTuple: (Int, String?)!
var tuple: (Int, String?)? {
set {
}
get {
return stubbedTuple
}
}
func method() {
}

File: UseCases/src/test/java/codes/seanhenry/mockgenerator/generator/templates/PropertyProtocolTest.kt
package codes.seanhenry.mockgenerator.generator.templates

import codes.seanhenry.mockgenerator.entities.Method
import codes.seanhenry.mockgenerator.entities.Property
import codes.seanhenry.mockgenerator.generator.MockTransformer

class PropertyProtocolTest : MockGeneratorTestTemplate() {
  override fun build(generator: MockTransformer) {
    generator.add(
        Property.Builder("readWrite")
            .type("String")
            .build(),
        Property.Builder("readOnly")
            .type("Int")
            .readonly()
            .build(),
        Property.Builder("optional")
            .type().optional { it.type("UInt") }
            .build(),
        Property.Builder("unwrapped")
            .type().optional { it.unwrapped().type("String") }
            .build(),
            Property.Builder ("tuple")
            .type().optional { opt ->
                  opt.type().tuple { tuple ->
                    tuple.element("Int")
                        .element().optional { it.type("String") }
                  }
                }
            .build()
        )
    generator.add(
        Method.Builder("method").build()
    )
  }
}
